packages/scaffold-config/src/ct-detect-third-party.ts

```diff
@@ -0,0 +1,104 @@
+import path from 'path'
+import { pathToFileURL } from 'url'
+import globby from 'globby'
+import { z } from 'zod'
+import Debug from 'debug'
+import type { ThirdPartyComponentFrameworkDefinition } from './frameworks'
+
+const debug = Debug('cypress:scaffold-config:ct-detect-thrid-party')
+
+const DependencySchema = z.object({
+  type: z.string(),
+  name: z.string(),
+  package: z.string(),
+  installer: z.string(),
+  description: z.string(),
+  minVersion: z.string(),
+})
+
+const DependencyArraySchema = z.array(DependencySchema)
+
+const BundlerSchema = z.enum(['webpack', 'vite'])
+
+const ThirdPartyComponentFrameworkSchema = z.object({
+  type: z.string().startsWith('cypress-ct-'),
+  configFramework: z.string().startsWith('cypress-ct-'),
+  name: z.string(),
+  supportedBundlers: z.array(BundlerSchema),
+  detectors: DependencyArraySchema,
+  dependencies: z.function(),
+  mountModule: z.function(),
+  componentIndexHtml: z.optional(z.function()),
+})
+
+const CT_FRAMEWORK_GLOB = 'node_modules/cypress-ct-*/package.json'
```

Comment:
Do we need something special for windows? Iirc some weirdness with globby and slashes?

Reply:
Good point I'll check